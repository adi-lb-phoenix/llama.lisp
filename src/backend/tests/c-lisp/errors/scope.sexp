(c-lisp
    (define ((print int) (n int)))

    (define ((main void))
        (declare i int)

        (for ((set i 0)
              (lt i 100)
              (set i (add i 1)))
            (declare val int)
            (if (eq
                    i
                    (mul (div i 7) 7))
                (set val i)))
        (call print val)))
