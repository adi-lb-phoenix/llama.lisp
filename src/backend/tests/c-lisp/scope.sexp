(c-lisp
    (define ((print int) (n int)))

    (define ((main void))
        (declare (sum1 int))
        (set sum1 0)
        (declare (sum2 int))
        (set sum2 0)

        (declare (same-name int))

        (for ((set same-name 0)
              (lt same-name 10)
              (set same-name (add same-name 1)))
            ((declare (same-name int))
             (set same-name 0)
             (set sum2 (add sum2 same-name)))
            (set sum1 (add sum1 same-name)))

        (call print sum1)
        (call print sum2)
        (ret)))
