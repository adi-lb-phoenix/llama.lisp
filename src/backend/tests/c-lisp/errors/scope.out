Error in statement: ['call', 'print', 'val']
CodegenError: Undeclared symbol: val
